src/lib.rs

```diff
@@ -216,6 +242,65 @@ impl Default for CertificateParams {
 	}
 }
 
+impl CertificateParams {
+	/// Parses the ca certificate from the ASCII PEM format
+	/// 
+	/// See `from_ca_cert_der` for more details.
+	#[cfg(all(feature = "pem", feature = "x509-parser"))]
+	pub fn from_ca_cert_pem(pem_str :&str, key_pair: KeyPair) -> Result<Self, RcgenError> {
+		let certificate = pem::parse(pem_str)
+			.or(Err(RcgenError::CouldNotParseCertificate))?;
+		let certificate_der :&[_] = &certificate.contents;
+		Self::from_ca_cert_der(certificate_der, key_pair)
+	}
+
+	/// Parses the ca certificate from the DER format
+	/// 
+	/// This function is only of use if you have an existing ca certificate with which
+	/// you want to sign a certificate newly generated by `rcgen` using the
+	/// `serialize_der_with_signer()` or `serialize_pem_with_signer()` functions.
+	/// 
+	/// Will not check if certificate is a ca certificate!
+	#[cfg(feature = "x509-parser")]
+	pub fn from_ca_cert_der(ca_cert: &[u8], key_pair: KeyPair) -> Result<Self, RcgenError> {
+		let (_remainder, x509) = x509_parser::parse_x509_der(ca_cert)
+			.or(Err(RcgenError::CouldNotParseCertificate))?;
+
+		// WDYT?
+		// if !x509.tbs_certificate.is_ca() {
```

Comment:
The other code doesn't do the check either. We make it clear enough that the parsing logic right now is so rudimentary that it only makes sense for signing other certs, so I think it's not needed.

Reply:
Removed